### PR TITLE
[SPO] Adding OData__UIVersionString document version field

### PR DESCRIPTION
**Pull request by llermaly**
This PR adds the `OData__UIVersionString` to the documents. This field is useful to filter out drafts using sync rules.

When this option enabled:

image

Files can have versions. 

0.1 means unpublished draft
After published becomes 1.0
Then I save a new draft and becomes 1.1

image

Sync rule to exclude drafts (everything that doesn't end with 0): `^(?!.*\.0$).*$`

#### Pre-Review Checklist
- [ ] this PR has a meaningful title
- [ ] this PR links to all relevant github issues that it fixes or partially addresses
- [ ] if there is no GH issue, please create it. Each PR should have a link to an issue
- [ ] this PR has a thorough description
- [ ] Covered the changes with automated tests
- [ ] Tested the changes locally
- [ ] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

**Comment by seanstory** [comment]
@llermaly can you sign the Contributor License Agreement? https://github.com/elastic/connectors/pull/1884#issuecomment-1803124369 

Then this should be good to merge.

**Comment by llermaly** [comment]
@seanstory signed!

**Comment by seanstory** [comment]
@llermaly I've rebased in such a way that I'm now seen as the only committer so that the CLA check passes.

**Comment by seanstory** [comment]
@elastic/ingestion-team can I get an approval? I rebased myself, so my approval doesn't count anymore 🙈

**Review by seanstory — APPROVED** [review]

**Pull request merged**
